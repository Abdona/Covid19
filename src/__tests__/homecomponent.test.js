import { React } from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import store from '../redux/configureStore';
import Home from '../components/homecomponent';
import '@testing-library/jest-dom';

describe('Test Home page Components', () => {
  it('test HomePage function component renders without error', () => {
    const divElement = document.createElement('div');
    ReactDOM.render(<Provider store={store}><Home /></Provider>, divElement);
  });

  it('render correctly', () => {
    const { getByTestId } = render(<Provider store={store}><Home /></Provider>);
    expect(getByTestId('homepage')).toHaveTextContent('Europe0STATS BY COUNTRY');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><Home /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
