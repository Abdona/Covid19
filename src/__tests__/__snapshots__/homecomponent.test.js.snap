// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home page Components matches snapshot 1`] = `
<div
  className="home"
  data-testid="homepage"
>
  <h3>
    Corona Statisitics
  </h3>
  <div
    className="country2"
    data-testid="country"
  >
    <div
      className="imgcont"
    >
      <img
        alt="Europe"
        src="country.svg"
      />
      
    </div>
    <div
      className="textcont"
    >
      <h1>
        Europe
      </h1>
      <h3>
        0
      </h3>
    </div>
  </div>
  <h3>
    STATS BY COUNTRY
  </h3>
  <div
    className="container"
  />
</div>
`;
